### ISSUE TITLE
Questions about causal methods

### ISSUE THREAD

**Issue by Tom-zgt** [LTH14/mar#27]
@LTH14   In my attempts with causal methods, I have managed to reduce the diffusion loss to 0.3, but the generated images during inference still turn out chaotic. I've noticed that after sampling, the values become extremely large. For example, the max and min values of the hidden states before sampling in simple diffusion are 4.65 and -3.51, respectively, but after sampling, they balloon to 2177.27 and -1183.81. Could this be due to insufficient training steps, or is there an issue with the sampling process?
hidden_stats:
tensor([[ 2.3809, -0.2876,  1.7492,  ...,  0.2722, -0.1037, -0.0582]],
       device='cuda:0')
tensor(-3.5128, device='cuda:0') tensor(4.6599, device='cuda:0')

the output of diffloss.sample
tensor([[ 1843.2513,   743.4045,  -769.0864,   699.2002,  -338.0317,   719.9684,
          -359.3054,  -290.0359,   340.5087,  -795.7183,   915.6506, -1262.3744,
           389.3547,  1107.4041,   241.5811,  -993.8033]], device='cuda:0')
tensor(-1262.3744, device='cuda:0') tensor(1843.2513, device='cuda:0')

generated image:
![000000_0_vae](https://github.com/user-attachments/assets/83baaf52-1dc9-47b1-83ed-09cda411582f)

**Comment by LTH14** [comment]
Since we use an ema of 0.9999, it takes around 100k steps to get reasonable generation performance.

**Comment by Tom-zgt** [comment]
> Since we use an ema of 0.9999, it takes around 100k steps to get reasonable generation performance.

Thanks for your reply. If we only focus on the loss (causal method), at what loss value would we start seeing normal images? Is it possible to train a model (can generate reasonable images) without using EMA?

**Comment by LTH14** [comment]
The loss is computed on the online model, which does not have ema. No ema can harm the performance, but it can still generate reasonable images -- you can just turn the use_ema off. However, it is hard to say what loss can ensure good generation performance, as the diffusion loss decreases slowly after the initial stage, but a slight decrease can result in a big change in generation performance. In general, the model should generate non-random images after 50k steps without ema.

**Comment by Tom-zgt** [comment]
> The loss is computed on the online model, which does not have ema. No ema can harm the performance, but it can still generate reasonable images -- you can just turn the use_ema off. However, it is hard to say what loss can ensure good generation performance, as the diffusion loss decreases slowly after the initial stage, but a slight decrease can result in a big change in generation performance. In general, the model should generate non-random images after 50k steps without ema.

Thanks very much for your patient answers.

**Comment by uanu2002** [comment]
@Tom-zgt I ran into the same problem, the diffusion output is very large, did you solve it, any tips?

**Comment by aiiph4** [comment]
@Tom-zgt @uanu2002 Same problem here. Any tips are appreciated!
